src/crimson/osd/main.cc

```diff
@@ -91,14 +139,22 @@ int main(int argc, char* argv[])
   using ceph::common::sharded_conf;
   using ceph::common::sharded_perf_coll;
   using ceph::common::local_conf;
+  ObsTracker config_tracker;
+  std::unique_ptr<ConfigValues> config_values = std::make_unique<ConfigValues>();
+  config_values->name = init_params.name;
+  config_values->cluster = cluster_name;
+  md_config_t ceph_config(*config_values, config_tracker, true);
+  ceph_config.parse_config_files(*config_values, config_tracker, conf_file_list.c_str(), &std::cerr, CODE_ENVIRONMENT_DAEMON);
+  ceph_config.parse_argv(*config_values, config_tracker, ceph_args);
+  if (maybe_daemonize(*config_values) < 0)
```

Comment:
@liu-chunmei after a second thought, i think probably the right approach is to drop the support of daemonize.

@liewegas @neha-ojha @athanatos what do you think?